index.js

```diff
@@ -2,19 +2,516 @@ const readline = require('readline');
 const readlineInterface = readline.createInterface(process.stdin, process.stdout);
 
 function ask(questionText) {
-  return new Promise((resolve, reject) => {
-    readlineInterface.question(questionText, resolve);
-  });
+	return new Promise((resolve, reject) => {
+		readlineInterface.question(questionText, resolve);
+	});
 }
 
-start();
+let playerInventory = [];
+
+//assigning variables as undefined in order to make global functions out of the scope of the room's
+
+collectedItem = undefined;
+userAnswer = undefined;
+droppableItems = undefined;
+roomItems = undefined;
+
+/*-----------global take function that will remove item from the room's inventory and place it into the players inventory*/
+let take = function (userAnswer) {
+	for (let items of roomInv) {
+		collectedItem = items;
+	}
+
+	for (let items of playerInventory) {
+		droppableItems = items;
+	}
+	if (roomInv.length <= 0) {
+		console.log("There's nothing here...");
+	} else if (immovableItemsTable.includes(userAnswer)) {
+		console.log(`I cant take this...${userAnswer}`);
+	} else if (roomInv.length > 0 && collectedItem.item === userAnswer) {
+		roomInv.splice(roomInv.indexOf(collectedItem));
+		playerInventory.push(collectedItem);
+		console.log(`You collected a ${collectedItem.item}!`);
+	} else {
+		console.log("I don't know what you want...");
+	}
+};
```

Comment:
Great set up for inventory management! Setting up a check on the room inventory length to verify that there is indeed an item still to take is a great catch. This successfully prevented me from taking items more than once, thus keeping the game from adding undefined items to the players inventory.